server/WebAPI/Services/UserService.cs

```diff
@@ -1,45 +1,70 @@
 using Entities;
 using Entities.Models;
-using System;
-using System.Security.Cryptography;
-using System.Text;
 
 namespace WebAPI.Services
 {
-    public class UserService
+  public class UserService
+  {
+
+    private readonly UserRepository _userRepository;
+
+    public UserService(UserRepository userRepository)
     {
+      _userRepository = userRepository;
+    }
 
-        private readonly UserRepository _userRepository;
-        
-        public UserService(UserRepository userRepository)
-        {
-            _userRepository = userRepository;
-        }
+    public User GetUser(string email, string password)
+    {
+      var user = _userRepository.FindUserByEmail(email);
+      if(user != null && CryptoUtils.VerifyPassword(password, user.Password))
+      {
+        return user;
+      }
+      return null;
+    }
 
-        public User GetUser(string email, string password)
-        {
-            var user = _userRepository.FindUserByEmail(email);
-            if (user != null && CryptoUtils.VerifyPassword(password, user.Password))
-            {
-                return user;
-            }
-            return null;
-        }
+    public bool IsUserExist(string email)
+    {
+      return _userRepository.FindUserByEmail(email) != null;
+    }
+
+    public void CreateUser(string email, string password)
+    {
+      var user = new User
+      {
+        Email = email,
+        Password = CryptoUtils.HashPassword(password)
+      };
+
+      _userRepository.CreateUser(user);
+    }
+    public User GetUser(string email)
+    {
+      return _userRepository.FindUserByEmail(email);
+    }
 
-        public bool IsUserExist(string email)
+    public bool UpdateUserPassword(string email, string newPassword)
```

Comment:
will this allow for attacker to change password for any user knowing only his email?

Reply:
no. it will not have a token that the server will use to skip the request

Reply:
lets assume i'm UserA, got my token, and handcrafted ajax request from frontend setting email of UserB here. 
as i'm legitimate user and have token, server will process this request. as there is no check if email is really email of UserA (current logged in user) so UserA can change password for any other user knowing only his email. isn't it?

Reply:
Yes, it is true.i rewrite